ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,244 @@
+            var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+            AddToSubscribersRepository(subscriberKey);
+            AddToCallBacksBySubscriberRepository(subscriberKey, callback);
+           
+            if (!string.IsNullOrEmpty(_privateCommunicationIdentifier))
+            {
+                InitPrivateSubscriber(component, stateMachine);
+            }
+            else
+            {
+                _xcObservable.Subscribe(callback);
+                Subscribe(component, stateMachine);
+            }
+        }
+
+        private void InitPrivateSubscriber(string component, string stateMachine)
+        {
+            _xcObservable.Subscribe(args =>
+            {
+                if (args.Header.StateMachineCode != _xcConfiguration.GetStateMachineCode(component, stateMachine) ||
+                    args.Header.ComponentCode != _xcConfiguration.GetComponentCode(component)) return;
+                var subscribreKey = new SubscriberKey(args.Header.ComponentCode, args.Header.StateMachineCode);
+                List<Action<MessageEventArgs>> callBackList;
+                if (!_callbacksBySubscriberKey.TryGetValue(subscribreKey, out callBackList)) return;
+                foreach (var callBack in callBackList)
+                {
+                    callBack(args);
+                }
+            });
+            Subscribe(component, stateMachine);
+        }
+
+        private void Subscribe(string component, string stateMachine)
+        {
+            var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+            RabbitMqSubscriberInfos consumerChannel;
+            _subscribers.TryGetValue(subscriberKey, out consumerChannel);
+            try
+            {
+                if (consumerChannel == null || consumerChannel.IsOpen) return;
+                if (_connection == null || !_connection.IsOpen) return;
+                var exchangeName = _xcConfiguration?.GetComponentCode(component).ToString();
+                var routingKey = _xcConfiguration?.GetSubscriberTopic(component, stateMachine);
+                var channel = _connection.CreateModel();
+                channel.ModelShutdown += ChannelOnModelShutdown;
+                channel.ExchangeDeclare(exchangeName, ExchangeType.Topic);
+                var queueName = channel.QueueDeclare().QueueName;
+                var subscriber = new EventingBasicConsumer(channel);
+                channel.BasicConsume(queueName, false, subscriber);
+                channel.QueueBind(queueName, exchangeName, routingKey, null);
+
+                var replyChannel = _connection.CreateModel();
+                replyChannel.ModelShutdown += ChannelOnModelShutdown;
+                        
+                var thread = new Thread(Listen);
```

Comment:
We are going to create too many threads. Check xcomponent implementation with mutualised consumer